lib/puppet/module.rb

```diff
@@ -58,7 +58,7 @@ def self.is_module_namespaced_name?(name)
   attr_accessor :dependencies, :forge_name
   attr_accessor :source, :author, :version, :license, :summary, :description, :project_page
 
-  def initialize(name, path, environment)
+  def initialize(name, path, environment, include_prerelease = true)
```

Comment:
Ideally, it should be controlled by a setting or option somewhere. The reason it is true here is that it resolves the PEZ problem and ensures backward compatibility for modules that might rely on the old behavior. So, pending a decision on how the more strict ranges should be handled from a migration perspective, I think it needs to be true for now.